File: StudentEmailGeneratorJava/FileHandler.java
import java.io.*;
import java.util.HashMap;
import java.util.Map;

public class FileHandler {
    private static final String STORAGE_FILE = "student_emails.txt";
    
    public void saveToFile(Map<String, Student> students) {
        try (PrintWriter writer = new PrintWriter(new FileWriter(STORAGE_FILE))) {
            for (Map.Entry<String, Student> entry : students.entrySet()) {
                String studentId = entry.getKey();
                Student student = entry.getValue();
                writer.println(studentId + "|" + student.getName() + "|" + student.getEmail());
            }
            System.out.println("Data saved to " + STORAGE_FILE);
        } catch (IOException e) {
            System.out.println("Error saving to file: " + e.getMessage());
        }
    }
    
    public Map<String, Student> loadFromFile() {
        Map<String, Student> students = new HashMap<>();
        
        File file = new File(STORAGE_FILE);
        if (!file.exists()) {
            return students;
        }
        
        try (BufferedReader reader = new BufferedReader(new FileReader(STORAGE_FILE))) {
            String line;
            while ((line = reader.readLine()) != null) {
                line = line.trim();
                if (!line.isEmpty()) {
                    String[] parts = line.split("\\|");
                    if (parts.length == 3) {
                        String studentId = parts[0];
                        String name = parts[1];
                        String email = parts[2];
                        students.put(studentId, new Student(name, email));
                    }
                }
            }
            System.out.println("Loaded " + students.size() + " student(s) from " + STORAGE_FILE);
        } catch (IOException e) {
            System.out.println("Error loading from file: " + e.getMessage());
        }
        
        return students;
    }
}

File: StudentEmailGeneratorPython/file_handler.py
import os

STORAGE_FILE = "student_emails.txt"


def save_to_file(students):
    try:
        with open(STORAGE_FILE, 'w') as file:
            for student_id, data in students.items():
                file.write(f"{student_id}|{data['name']}|{data['email']}\n")
        print(f"Data saved to {STORAGE_FILE}")
    except Exception as e:
        print(f"Error saving to file: {e}")


def load_from_file():
    students = {}
    
    if not os.path.exists(STORAGE_FILE):
        return students
    
    try:
        with open(STORAGE_FILE, 'r') as file:
            for line in file:
                line = line.strip()
                if line:
                    parts = line.split('|')
                    if len(parts) == 3:
                        student_id, name, email = parts
                        students[student_id] = {
                            'name': name,
                            'email': email
                        }
        print(f"Loaded {len(students)} student(s) from {STORAGE_FILE}")
    except Exception as e:
        print(f"Error loading from file: {e}")
    
    return students